packages/drivekit_trip_analysis/flutter_drivekit_trip_analysis_android/lib/src/adapter.dart

```diff
@@ -27,3 +27,466 @@ extension PigeonAdapter on Vehicle {
+  PigeonLogbook toPigeonImplementation() {
+    return PigeonLogbook(
+      status: status,
+      updateDate: updateDate,
+    );
+  }
+}
+
+/// Adapts the [SpeedingStatistics] class to the Pigeon
+extension SpeedingStatisticsAdapter on SpeedingStatistics {
+  /// Converts a [PigeonSpeedingStatistics] to a corresponding Pigeon
+  PigeonSpeedingStatistics toPigeonImplementation() {
+    return PigeonSpeedingStatistics(
+      distance: distance,
+      duration: duration,
+      speedingDistance: speedingDistance,
+      speedingDuration: speedingDuration,
+      score: score,
+      speedLimitContexts: speedLimitContexts
+          .map((SpeedLimitContext e) => e.toPigeonImplementation())
+          .toList(),
+    );
+  }
+}
+
+/// Adapts the [AdvancedSafety] class to the corresponding Pigeon
+extension AdvancedSafetyAdapter on AdvancedSafety {
+  /// Converts a [PigeonAdvancedSafety] to a corresponding model class.
+  PigeonAdvancedSafety toPigeonImplementation() {
+    return PigeonAdvancedSafety(
+      safetyContext: safetyContext
+          .map((SafetyContext e) => e.toPigeonImplementation())
+          .toList(),
+    );
+  }
+}
+
+/// Adapts the [AdvancedEcoDriving] class to the corresponding Pigeon
+extension AdvancedEcoDrivingAdapter on AdvancedEcoDriving {
+  /// Converts a [AdvancedEcoDriving] to a corresponding Pigeon
+  PigeonAdvancedEcoDriving toPigeonImplementation() {
+    return PigeonAdvancedEcoDriving(
+      ecoDrivingContext:
+          ecoDrivingContext.map((e) => e.toPigeonImplementation()).toList(),
+    );
+  }
+}
+
+/// Adapts the [AdvancedFuelEstimation] class to the corresponding Pigeon
+extension AdvancedFuelEstimationAdapter on AdvancedFuelEstimation {
+  /// Converts a [PigeonAdvancedFuelEstimation] to a corresponding model class.
+  PigeonAdvancedFuelEstimation toPigeonImplementation() {
+    return PigeonAdvancedFuelEstimation(
+      fuelEstimationContext:
+          fuelEstimationContext.map((e) => e.toPigeonImplementation()).toList(),
+    );
+  }
+}
+
+/// Adapts the [Route] class to the corresponding Pigeon
+extension RouteAdapter on Route {
+  /// Converts a [Route] to a corresponding Pigeon.
+  PigeonRoute toPigeonImplementation() {
+    return PigeonRoute(
+      gpsDate: gpsDate.toList(),
+      gpsVelocity: gpsVelocity.toList(),
+      longitude: longitude.toList(),
+      latitude: latitude.toList(),
+      gpsElevation: gpsElevation.toList(),
+      gpsAccuracy: gpsAccuracy.toList(),
+      gpsHeading: gpsHeading.toList(),
+      screenLocked: screenLocked.toList(),
+      activityValue: activityValue.toList(),
+      roll: roll.toList(),
+      pitch: pitch.toList(),
+      yaw: yaw.toList(),
+      gyroscopeNormVar: gyroscopeNormVar.toList(),
+    );
+  }
+}
+
+/// Adapts the [PostGenericResponse] class to the corresponding Pigeon
+extension PigeonPostGenericResponseAdapter on PostGenericResponse {
+  /// Converts a [PigeonPostGenericResponse] to a corresponding model class.
```

Comment:
"/// Converts a [**PostGenericResponse**] to a corresponding **Pigeon**"

Reply:
corrigé sur 7cecc2affa21960d1b1c86074dea479cc4822f35